identity/identity.go

```diff
@@ -160,6 +160,24 @@ func (t *Traits) String() string {
 	return string(*t)
 }
 
+// Merge merges the other traits into this one. If a key exists in both, the
+// other traits' value is use
+func (t *Traits) Merge(o Traits) (Traits, error) {
```

Comment:
This only merges the first level. Is that what we want? If not, this should be something like mapstructure.Merge or similar.

Reply:
Also test please